fix:  Crash while running command

### Is there an existing issue for this?

- [X] I have searched the existing issues.

### Version

2.7.1

### Description

Melos crashes on CI, with logs: 

```
Unhandled exception:
Null check operator used on a null value
#0      _RunMixin._runScript (package:melos/src/commands/run.dart:136:[10]
<asynchronous suspension>
#1      _RunMixin.run (package:melos/src/commands/run.dart:23:22)
<asynchronous suspension>
#2      RunCommand.run (package:melos/src/command_runner/run.dart:53:14)
<asynchronous suspension>
#3      CommandRunner.runCommand (package:args/command_runner.dart:209:[13]
<asynchronous suspension>
#4      MelosCommandRunner.runCommand (package:melos/src/command_runner.dart:80:5)
<asynchronous suspension>
#5      main (file:///Users/runner/hostedtoolcache/flutter/stable-3.3.3-x64/.pub-cache/hosted/pub.dartlang.org/melos-2.7.1/bin/melos.dart:44:5)
<asynchronous suspension>
Error: Process completed with exit code 255.
```

the command is this one: 

`melos exec --fail-fast -- \
      flutter pub run build_runner build --delete-conflicting-outputs`

### Steps to reproduce

1. run the command
2. see the crash

### Expected behavior

should not crash

### Screenshots

_No response_

### Additional context and comments

_No response_

--- a/packages/melos/lib/src/commands/run.dart
+++ b/packages/melos/lib/src/commands/run.dart
@@ -63,7 +63,7 @@ mixin _RunMixin on _Melos {
       requirePrompt: true,
     );
 
-    final selectedScriptIndex = scriptChoices.indexOf(selectedScript!);
+    final selectedScriptIndex = scriptChoices.indexOf(selectedScript);
 
     return scripts[selectedScriptIndex].name;
   }
@@ -133,7 +133,7 @@ mixin _RunMixin on _Melos {
           choices,
           interactive: false,
           defaultsTo: choices[0],
-        )!;
+        );
       }
 
       final selectedPackageIndex =

--- a/packages/melos/lib/src/common/utils.dart
+++ b/packages/melos/lib/src/common/utils.dart
@@ -220,7 +220,7 @@ bool promptBool({
   );
 }
 
-T? promptChoice<T>(
+T promptChoice<T>(
   String message,
   Iterable<T> options, {
   T? defaultsTo,
@@ -236,7 +236,7 @@ T? promptChoice<T>(
       defaultsTo: defaultsTo,
       prompt: prompt,
       interactive: interactive,
-    ),
+    )!,
     message: message,
     defaultsTo: defaultsTo,
     defaultsToWithoutPrompt: defaultsToWithoutPrompt,

